src/c_kzg_4844.c

```diff
@@ -3383,25 +3391,37 @@ static C_KZG_RET verify_kzg_proof_multi_impl(
         blst_fr_mul(&inv_h_pow, &inv_h_pow, &inv_h);
     }
 
-    /* Compute [x^n] in G_2 */
+    /* Commit to the interpolation polynomial, i.e., get [I(tau)] in G_1 */
+    ret = poly_to_kzg_commitment_monomial(
+        &interpolation_poly_g1, interpolation_poly, n, s
+    );
+    if (ret != C_KZG_OK) goto out;
+
+    ///////////////////////////////////////////////////////////////////////////
+    // STEP 2: Compute the vanishing polynomial V(X) (as a commitment in G_2).
+    //         It is zero over the coset. In our case: V(X) = X^n - h^n
+    ///////////////////////////////////////////////////////////////////////////
+
+    /* Compute [h^n] in G_2 */
     blst_fr_eucl_inverse(&h_pow, &inv_h_pow);
     g2_mul(&h_pow_g2, blst_p2_generator(), &h_pow);
 
-    /* Compute [s^n - h^n] in G_2 */
-    g2_sub(&s_pow_minus_h_pow, &s->g2_values_monomial[n], &h_pow_g2);
+    /* Compute [V(tau)] = [tau^n - h^n] in G_2 */
```

Comment:
Slight preference for `s` over `tau` because that's what many Ethereum-related documents have been using. It also looks a bit better in code docs because it's a single letter. No strong opinion.

Reply:
I did not want to use `s` because the KZGSettings are called `s` everywhere. But I can call it `s` if you think it makes sense. What do you think?

Reply:
Reasonable point. Let's keep it as tau. We can change it later if we want.